crates/types/src/messages.rs

```diff
@@ -1,25 +1,27 @@
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::SerializeHierarchy;
 use spl_network_messages::{GameControllerReturnMessage, GameControllerStateMessage, SplMessage};
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Deserialize, Serialize, SerializeHierarchy)]
 pub enum IncomingMessage {
     GameController(GameControllerStateMessage),
     Spl(SplMessage),
 }
 
 impl Default for IncomingMessage {
     fn default() -> Self {
-        IncomingMessage::GameController(Default::default())
+        Self::Spl(Default::default())
     }
 }
```

Comment:
We had this already in Framework v2, so only a minor change in this PR. There is already a task for Ideas for next iteration because this needs a lot of refactoring and is therefore out-of-scope for this PR.